sympy/vector/implicitregion.py

```diff
@@ -107,6 +123,121 @@ def regular_point(self):
 
         raise NotImplementedError()
 
+    def _regular_point_parabola(self, a, b, c, d, e, f):
+            ok = (a, d) != (0, 0) and (c, e) != (0, 0) and b**2 == 4*a*c and (a, c) != (0, 0)
+
+            if not ok:
+                raise ValueError("Rational Point on the conic does not exist")
+
+            if a != 0:
+                d_dash, f_dash = (4*a*e - 2*b*d, 4*a*f - d**2)
+                if d_dash != 0:
+                    y_reg = -f_dash/d_dash
+                    x_reg = -(d + b*y_reg)/(2*a)
+                else:
+                    ok = False
+            elif c != 0:
+                d_dash, f_dash = (4*c*d - 2*b*e, 4*c*f - e**2)
+                if d_dash != 0:
+                    x_reg = -f_dash/d_dash
+                    y_reg = -(e + b*x_reg)/(2*c)
+                else:
+                    ok = False
+
+            if ok:
+                return x_reg, y_reg
+            else:
+                raise ValueError("Rational Point on the conic does not exist")
+
+    def _regular_point_ellipse(self, a, b, c, d, e, f):
+            D = 4*a*c - b**2
+            ok = D
+
+            if not ok:
+                raise ValueError("Rational Point on the conic does not exist")
+
+            if a == 0 and c == 0:
+                K = -1
+                L = 4*(d*e - b*f)
+            elif c != 0:
+                K = D
+                L = 4*c**2*d**2 - 4*b*c*d*e + 4*a*c*e**2 + 4*b**2*c*f - 16*a*c**2*f
+            else:
+                K = D
+                L = 4*a**2*e**2 - 4*b*a*d*e + 4*b**2*a*f
+
+            ok = L != 0 and not(K > 0 and L < 0)
+            if not ok:
+                raise ValueError("Rational Point on the conic does not exist")
+
+            K = Rational(K).limit_denominator(10**12)
+            L = Rational(L).limit_denominator(10**12)
+
+            k1, k2 = K.p, K.q
+            l1, l2 = L.p, L.q
+            g = gcd(k2, l2)
+
+            a1 = (l2*k2)/g
+            b1 = (k1*l2)/g
+            c1 = -(l1*k2)/g
+            a2 = sign(a1)*core(abs(a1), 2)
+            r1 = sqrt(a1/a2)
+            b2 = sign(b1)*core(abs(b1), 2)
+            r2 = sqrt(b1/b2)
+            c2 = sign(c1)*core(abs(c1), 2)
+            r3 = sqrt(c1/c2)
+
+            g = gcd(gcd(a2, b2), c2)
+            a2 = a2/g
+            b2 = b2/g
+            c2 = c2/g
+
+            g1 = gcd(a2, b2)
+            a2 = a2/g1
+            b2 = b2/g1
+            c2 = c2*g1
+
+            g2 = gcd(a2,c2)
+            a2 = a2/g2
+            b2 = b2*g2
+            c2 = c2/g2
+
+            g3 = gcd(b2, c2)
+            a2 = a2*g3
+            b2 = b2/g3
+            c2 = c2/g3
+
+            x, y, z = symbols("x y z")
+            eq = a2*x**2 + b2*y**2 + c2*z**2
+
+            sol = diophantine(eq)
+            sol = list(sol)[0]
+            syms = Tuple(*sol).free_symbols
+            rep = {s: 3 for s in syms}
+
```

Comment:
There is a bug here. In cases like `x*y = 1`, the solution of diophantine equation is {(p**2 + q**2, -2*p*q, p**2 - q**2)}. If I substitute p = 3 and q = 3, z turns out to be zero. This leads to the result being nan. The algorithm requires the solution of diophantine equation to be non-trivial.

Reply:
What about pushing the solution of the diophantine equation into the equation solver? For example, in the case `{(p2 + q2, -2pq, p2 - q2)}`, can it help if you get `p` as a function of `q`?

Reply:
Yes, I think this should work. Solving for z != 0, then using one of its solutions.

Reply:
This is how I am trying to get the solutions for z != 0
```python
for sol in solutions:
    syms = Tuple(*sol).free_symbols
    
    z = sol[2]
    
    z_syms = ztemp.free_symbols

    if len(z_syms) == 2:
        print(solveset(Unequality(z, 0), next(iter(z_syms)), S.Integers))
```

But solveset is too slow for returning an Integer solution.

Reply:
Have you thought about solving the equality and then taking the complement of the solutions? It should be equivalent to `solveset(Unequality( ... ))`.

Reply:
This should work. In the case of two variables p and q, I will iterate p over S.Integers and then solve q for z != 0.

Reply:
ping @Upabjojr.